src/sentry/api/endpoints/auth_index.py

```diff
@@ -139,8 +142,8 @@ def _validate_superuser(
         SSO and if they do not, we redirect them back to the SSO login.
 
         """
-        # TODO Look at AuthVerifyValidator
-        validator.is_valid()
+        if not validator.is_valid():
+            raise NotAuthenticated(detail=MISSING_AUTH_ERROR_MESSAGE)
```

Comment:
should this raise `validator.errors`?

Reply:
I want to return 401 here instead of the default 400 that comes with `serializer.errors`. I'll change it to use detail code instead of detail to match the rest of the file tho

Reply:
`validator` is a serializer though

Reply:
it's validating that you have a password or u2f challenge/response, so I think it has more to do with auth'ing than a bad request body

Reply:
This feels like more of a 400 error to me. The validator isn't actually validating whether those fields are correct, just that they exist. 

A 401 implies that an authorization check happened, which isn't happening here.

Reply:
hmm ok I can switch to having the serializer raise a validation error with a code and then return `serializer.errors`